### PR TITLE
fixed issues related to CRA 4

### LINKED ISSUE #248: react-scripts v4 support?
Will this project be updated to support react-scripts v4?

When running with react-scripts v4, I get the following error:

```
my-project/node_modules/cra-build-watch/scripts/index.js:55
config.entry = config.entry.filter(fileName => !fileName.match(/webpackHotDevClient/));
                            ^

TypeError: config.entry.filter is not a function
    at Object.<anonymous
```

### PR DESCRIPTION
**Pull request by tavikukko**
Just fixed issues when using with CRA 4, but there might be some breaking changes that need attention too.
fixes #248

### CODE REVIEW

File: scripts/index.js [hunk 1]
```diff
@@ -52,7 +52,9 @@ const env = getClientEnvironment(process.env.PUBLIC_URL || ''); // eslint-disabl
 /**
  * We need to update the webpack dev config in order to remove the use of webpack devserver
  */
-config.entry = config.entry.filter(fileName => !fileName.match(/webpackHotDevClient/));
+if (major < 4) {
+  config.entry = config.entry.filter(fileName => !fileName.match(/webpackHotDevClient/));
+}
 config.plugins = config.plugins.filter(
   plugin => !(plugin instanceof webpack.HotModuleReplacementPlugin)
```
Comment (keremciu) [inline scripts/index.js:59]: Error: [React Refresh] Hot Module Replacement (HMR) is not enabled! React Refresh requires HMR to function properly.

I tried to use it in my game but gave me the error above you can find the code of the game here: https://github.com/keremciu/house-of-decks/blob/ws-implement/client/package.json

if I run `FAST_REFRESH=false cra-build-watch` this way it works but I'd like to learn how react-refresh-webpack plugin is affecting process

Comment (keremciu) [reply scripts/index.js:59]: how about wrapping this filter function with process.env.FAST_REFRESH?

Comment (tavikukko) [reply scripts/index.js:59]: why do you need to use cra-build-watch in your project?

Comment (tavikukko) [reply scripts/index.js:60]: Finally got the same error as you, updated the PR, I disabled the fast refresh because its not needed with this.

Comment (keremciu) [reply scripts/index.js:60]: I use it for development, basically I run `nodemon` on server then I needed to watch my changes on client side too.

I can confirm by disabling FAST_REFRESH, it works 💯

Comment (bjankord) [reply scripts/index.js:60]: Tested these changes in my app that uses cra-build-watch and these worked with react-scripts v4. :+1:

**Comment by NicholasTuck** [comment]
Just started a new CRA, which uses react-scripts@4.0.0. We had the same error that this is trying to resolve, we pointed to the branch, and it worked for us.  👍  Thank you!

**Comment by bjankord** [comment]
@Nargonath Does this PR look ok? Do you think we could get this merged and a release cut on this?

**Review by Nargonath — APPROVED** [review]
I took the liberty to revert the commits about the dependencies update. I prefer to update the dep one by one and test their impact if any that way it is better documented in the changelog too. Also some dependencies can have breaking changes in the node version they support potentially breaking cra-build-watch node version compat.

Thank you everyone involved in this. Your help is definitely appreciated. 👍

File: scripts/index.js (hunk 1 above)
Comment (Nargonath) [reply scripts/index.js:60]: Thanks for investigating this.

**Review by Nargonath — APPROVED** [review]

**Comment by Nargonath** [comment]
There is an issue with semantic release and new npm automation token right now. You can follow along here: https://github.com/Nargonath/cra-build-watch/issues/256. Once this issue is fixed I can proceed with the release.

**Pull request merged**